spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -165,7 +166,11 @@ private Schema snapshotSchema() {
   @Override
   public StructType schema() {
     if (lazyTableSchema == null) {
-      this.lazyTableSchema = SparkSchemaUtil.convert(snapshotSchema());
+      if (icebergTable instanceof BaseMetadataTable) {
+        this.lazyTableSchema = SparkSchemaUtil.convert(icebergTable.schema());
```

Comment:
So this seems to disable time-travel using the right schema for Metadata Tables.  I think we should fix in long run, but maybe this is ok for a quick fix for now. 

How about, we can put this logic in SnapshotUtil.schemaFor().  If table is BaseMetadataTable, return current the schema.  That way, I think we can change just one place in the code?  And fix it there later when we want to implement this?